datafusion/execution/src/memory_pool/pool.rs

```diff
@@ -231,6 +235,11 @@ impl MemoryPool for FairSpillPool {
     }
 }
 
+/// Constructs a resources error based upon the individual [`MemoryReservation`].
+///
+/// The error references the `bytes already allocated` for the reservation,
+/// and not the total within the collective [`MemoryPool`],
+/// nor the total across multiple reservations with the same [`MemoryConsumer`].
 #[inline(always)]
 fn insufficient_capacity_err(
```

Comment:
In a follow up PR, I would like to iterate on this error message.

Take the original/current:
`Failed to allocate additional {} bytes for {} with {} bytes already allocated - maximum available is {}`

And change into something like:
`Failed to allocate additional {} bytes for {} with {} bytes already allocated for this reservation - {} bytes remain available for the total pool`

Reply:
This^^ suggestion is because this error message is inherently about only the reservation itself. Each reservation can request bytes from the pool -- but the pool itself only incr/decrements totals for the pool. It does **NOT** track each reservation, nor if multiple reservations have the same consumer.

As a result, the simplest solution is to: (a) update this error message, and (b) continue to rely upon each reservation to resize itself to upon drop ([as it does](https://github.com/apache/datafusion/blob/01dc3f9766e1e36ef9864effdb1a7c89d138a982/datafusion/execution/src/memory_pool/mod.rs#L307-L310)).

Reply:
I like the idea of updating the error message in a follow on PR